src/storage/mvcc/reader/reader.rs

```diff
@@ -124,6 +124,9 @@ pub struct MvccReader<S: EngineSnapshot> {
     lock_cursor: Option<Cursor<S::Iter>>,
     write_cursor: Option<Cursor<S::Iter>>,
 
+    lower_bound: Option<Key>,
```

Comment:
Why not use `ForwardScanner`? It can use customized boundaries.

Reply:
Why we can use it? It cannot use snapshot.

Reply:
It can, otherwise it won't be defined as `struct ForwardScanner<S: Snapshot, P: ScanPolicy<S>>`.

Reply:
I don't think we should use ForwardScanner here, GC is highly combined with MvccReader.

Reply:
Then I suggest to remove these boundaries. It's limited by the region ranges already.

Reply:
The boundaries is used to avoid scanning tombstones unnecessarily. See https://github.com/tikv/tikv/pull/11922.